src/link/error/index.ts

```diff
@@ -44,9 +44,14 @@ export function onError(errorHandler: ErrorHandler): ApolloLink {
       try {
         sub = forward(operation).subscribe({
           next: (result) => {
-            if (result.errors) {
+            const handler = operation.client["queryManager"].incrementalHandler;
+            const errors =
+              handler?.isIncrementalResult(result) ?
+                handler.extractErrors(result)
+              : (result as FormattedExecutionResult).errors;
```

Comment:
```suggestion
              : result.errors;
```

Thanks to the type narrowing from `handler.isIncrementalResult`, it doesn't look like you need this type cast.